crypto/rand/drbg_lib.c

```diff
@@ -334,54 +481,64 @@ void *RAND_DRBG_get_ex_data(const RAND_DRBG *drbg, int idx)
  */
 
 /*
- * Creates a global DRBG with default settings.
+ * Initializes the DRBG with default settings.
+ * For global DRBG's a global lock is created with the given name
  * Returns 1 on success, 0 on failure
  */
-static int setup_drbg(RAND_DRBG *drbg, const char *name)
+static int drbg_setup(RAND_DRBG *drbg, const char *name)
 {
     int ret = 1;
 
-    drbg->lock = CRYPTO_THREAD_glock_new(name);
-    ret &= drbg->lock != NULL;
-    drbg->size = RANDOMNESS_NEEDED;
-    drbg->secure = CRYPTO_secure_malloc_initialized();
-    /* If you change these parameters, see RANDOMNESS_NEEDED */
+    if (name != NULL) {
+        OPENSSL_assert(drbg->lock == NULL);
```

Comment:
Some might complain of the new usage of OPENSSL_assert() here (since its behavior is capable of terminating the program on condition failure even in NDEBUG builds), but this truly is a "cannot happen" scenario with the current code.

Reply:
Removed `OPENSSL_assert()` and added `RANDerr()` calls. Please re-review `drbg_setup()`

Reply:
I am not seeing the new changes yet; did your push fail?

Reply:
Sorry, I had some errors to fix and as a consequence didn't manage to push before having to leave for an appointment.